internal/aws/awsutil/conn.go

```diff
@@ -383,3 +356,51 @@ func getPartition(region string) string {
 	p, _ := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region)
 	return p.ID()
 }
+
+// order
+// 1. override creds providers
+// 2. shared creds file
+// 3. shared profile
+// explicit keys do not make sense in this context since contrib does not take in explicit keys
+// and cwa does not provide explicit keys to config but come from these methods of getting creds
+// @TODO make this more upstream friendly working in the future
+func customCredentialProvider(cfg *AWSSessionSettings) []credentials.Provider {
+	overrideCredProviders := override.GetCredentialsChainOverride().GetCredentialsChain()
+	credProviders := make([]credentials.Provider, 0)
+	for _, provider := range overrideCredProviders {
+		for _, file := range cfg.SharedCredentialsFile {
+			credProviders = append(credProviders, provider(file))
+		}
+	}
+	// DO NOT TRY TO OPTIMIZE THIS CODE !!!
+	// If you do please do not change order
+	// this is to keep ordering where if there is a profile we want to try those creds first
+	if cfg.Profile != "" && cfg.SharedCredentialsFile != nil {
+		for _, file := range cfg.SharedCredentialsFile {
+			credProviders = append(credProviders, &RefreshableSharedCredentialsProvider{
+				sharedCredentialsProvider: &credentials.SharedCredentialsProvider{
+					Filename: file,
+					Profile:  cfg.Profile,
+				},
+			})
+		}
+	} else if cfg.Profile != "" {
+		credProviders = append(credProviders, &RefreshableSharedCredentialsProvider{
+			sharedCredentialsProvider: &credentials.SharedCredentialsProvider{
+				Filename: "",
+				Profile:  cfg.Profile,
+			},
+		})
+	}
+	if cfg.Profile == "" && cfg.SharedCredentialsFile != nil {
```

Comment:
nit: should this be `else if`?

Im ok with it for now, but this code will be confusing when I look at it in the future.
Wonder if it can be simplified to just create two vars for profile and files. files can be initialized using a for loop when cfg.SharedCredentialsFile is not nil and initialized to a slice with "" when it is nil.
Looks like we always set cfg.Profile even if its "".